lib/elixir/lib/module/parallel_checker.ex

```diff
@@ -175,12 +218,12 @@ defmodule Module.ParallelChecker do
     state
   end
 
-  defp spawn_checkers(%{modules: [{map, binary} | modules]} = state) do
+  defp spawn_checkers(%{modules: [{map, chunks} | modules]} = state) do
     parent = self()
     ets = state.ets
 
     spawn_link(fn ->
-      {binary, warnings} = Module.Checker.verify(map, binary, {parent, ets})
+      {binary, warnings} = Module.Checker.verify(map, chunks, {parent, ets})
```

Comment:
I am a bit worried about the amount of data we are handling here. We are passing the chunks to the child process only for it to pass it here which is then passed to the client. I think we can improve this in a couple ways.

1. Only send the chunk up here. We already have the whole binary, we can only receive the chunk and do the rest of the work

2. Skip sending the results to the parallel checker and instead send it directly to the process that started the parallel checker (the one we return warnings and binaries too). Both warnings and binaries can be collected directly there. This means that the process that starts the parallel checker should: a. have a simple receive loop to collect binaries and warnings b. monitor the parallel checker and finish when the parallel checker terminates. This means the parallel checker does coordination and caching but it doesn't collect results.

Another positive aspect of these two changes is that we don't need to send the binary to the parallel checker of compiled modules. We will either send them the map_definition (compiled modules) or the binary (runtime modules).

Reply:
Please check if this is what you had in mind.